spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AppConfigurationBootstrapConfiguration.java

```diff
@@ -67,31 +66,50 @@ public CloseableHttpClient closeableHttpClient() {
 
     @Bean
     public AppConfigurationPropertySourceLocator sourceLocator(AppConfigurationProperties properties,
-            AppConfigurationProviderProperties appProperties, ClientStore clients, ApplicationContext context) {
+            AppConfigurationProviderProperties appProperties, ClientStore clients, ApplicationContext context,
+            Optional<KeyVaultCredentialProvider> keyVaultCredentialProviderOptional,
+            Optional<SecretClientBuilderSetup> keyVaultClientProviderOptional) {
```

Comment:
It looks like not good practice to use Optionals as public method arguments. This forces the client used to add extra logics and unnecessary complexity. I like this article that talks about it http://dolszewski.com/java/java-8-optional-use-cases/.
If we want to be protected against possible NPE what do you think if we have the optional use in another way such as

```java 
public AppConfigurationPropertySourceLocator sourceLocator(AppConfigurationProperties properties,
            AppConfigurationProviderProperties appProperties, 
            ClientStore clients, 
            ApplicationContext context,
            KeyVaultCredentialProvider keyVaultCredentialProvider,
            SecretClientBuilderSetup keyVaultClientProvider) {
            
      Optional<KeyVaultCredentialProvider> keyVaultCredentialProviderOptional = offNullable(keyVaultCredentialProvider);
      Optional<SecretClientBuilderSetup> keyVaultClientProviderOptional = offNullable(keyVaultClientProvider);
....
```